### PR TITLE
add https readme 

### PR DESCRIPTION
**Pull request by clyang82**
Signed-off-by: Chun Lin Yang <clyang@cn.ibm.com>

Add samples/https/README file to tell user how to use it.

### CODE REVIEW

File: samples/https/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+This sample consists of an NGINX service which can encrypt traffic using HTTPS.
+
+To use it:
+
+1. Generate certificates
+
+    You need to have openssl installed to run these commands:
+
+    `openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout /tmp/nginx.key -out /tmp/nginx.crt -subj "/CN=my-nginx/O=my-nginx"`
```
Comment (venilnoronha) [inline samples/https/README.md]:
```suggestion
    ```console
    $ openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout /tmp/nginx.key -out /tmp/nginx.crt -subj "/CN=my-nginx/O=my-nginx"
    ```
```

File: samples/https/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+    `kubectl create secret tls nginxsecret --key /tmp/nginx.key --cert /tmp/nginx.crt`
```
Comment (venilnoronha) [inline samples/https/README.md]:
```suggestion
    ```console
    $ kubectl create secret tls nginxsecret --key /tmp/nginx.key --cert /tmp/nginx.crt
    ```
```

File: samples/https/README.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+2. Create a configmap used for the HTTPS service
+
+    `kubectl create configmap nginxconfigmap --from-file=samples/https/default.conf`
```
Comment (venilnoronha) [inline samples/https/README.md]:
```suggestion
    ```console
    $ kubectl create configmap nginxconfigmap --from-file=samples/https/default.conf
    ```
```

File: samples/https/README.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+1. Generate certificates
+
+    You need to have openssl installed to run these commands:
+
+    `openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout /tmp/nginx.key -out /tmp/nginx.crt -subj "/CN=my-nginx/O=my-nginx"`
+
+    `kubectl create secret tls nginxsecret --key /tmp/nginx.key --cert /tmp/nginx.crt`
+
+2. Create a configmap used for the HTTPS service
```
Comment (venilnoronha) [inline samples/https/README.md]:
```suggestion
2. Create a `ConfigMap` used for the HTTPS service
```

File: samples/https/README.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+    `kubectl create configmap nginxconfigmap --from-file=samples/https/default.conf`
+
+3. Deploy a NGINX-based HTTPS service
+
+    `kubectl apply -f samples/https/nginx-app.yaml`
```
Comment (venilnoronha) [inline samples/https/README.md]:
```suggestion
    ```console
    $ kubectl apply -f samples/https/nginx-app.yaml
    ```
```

File: samples/https/README.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
+# HTTPS service
+
+This sample consists of an NGINX service which can encrypt traffic using HTTPS.
+
+To use it:
+
+1. Generate certificates
+
+    You need to have openssl installed to run these commands:
```
Comment (venilnoronha) [inline samples/https/README.md]:
```suggestion
    You need to have [OpenSSL](https://www.openssl.org/) installed to run these commands:
```

File: samples/https/README.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+2. Create a configmap used for the HTTPS service
+
+    `kubectl create configmap nginxconfigmap --from-file=samples/https/default.conf`
+
+3. Deploy a NGINX-based HTTPS service
+
+    `kubectl apply -f samples/https/nginx-app.yaml`
+
+Call my-nginx service `curl https://my-nginx -k` to verify this creation.
```
Comment (venilnoronha) [inline samples/https/README.md]:
```suggestion
Send a request to the `my-nginx` service over HTTPS using the following command.

    ```console
    $ curl https://my-nginx -k
    ```
```

File: samples/https/README.md (hunk 6 above)
Comment (sdake) [reply samples/https/README.md]: You need to install [OpenSSL](https://www.openssl.org/) if not already installed:

File: samples/https/README.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
+# HTTPS service
+
+This sample consists of an NGINX service which can encrypt traffic using HTTPS.
```
Comment (vadimeisenbergibm) [inline samples/https/README.md]: I would remove which can encrypt traffic and leave just: _This sample defines an HTTPS service backed by NGINX._

Comment (clyang82) [reply samples/https/README.md]: fixed. Thanks.

**Comment by linsun** [comment]
@clyang82 @venilnoronha are comments from @vadimeisenbergibm and @frankbu being addressed now?

**Comment by clyang82** [comment]
> @clyang82 @venilnoronha are comments from @vadimeisenbergibm and @frankbu being addressed now?

Not being addressed. Let me find some time this week to address the comments.

**Comment by clyang82** [comment]
not stale

File: samples/https/README.md (hunk 8 above)
Comment (howardjohn) [reply samples/https/README.md]: Doesn't seem fixed?

**Comment by howardjohn** [comment]
Can you rebase your change tests can run again? Also this will be easier to get merged in on the master branch if you want to reopen there

**Comment by duderino** [comment]
@clyang82 looks like this one died a silent death.   Sorry it didn't make it into 1.1.  Agree with others on the review who suggested it should go to the istio/istio.io repository

**Pull request closed without merging**